noobaa_sa/s3_client.py

```diff
@@ -177,9 +257,21 @@ def get_object(self, bucket_name, object_key):
                 - "ContentLength": the size of the object in bytes
                 - "ResponseMetadata": a dict containing the response metadata
 
+        Raises:
+            NoSuchKey: If the object does not exist
+            ClientError: If an unexpected error occurs
+
         """
         log.info(f"Getting object {object_key} from bucket {bucket_name} via boto3")
-        output = self._boto3_client.get_object(Bucket=bucket_name, Key=object_key)
+        try:
+            output = self._boto3_client.get_object(Bucket=bucket_name, Key=object_key)
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "NoSuchKey":
+                log.warn(f"Object {object_key} does not exist in bucket {bucket_name}")
+                raise NoSuchKey(e)
+            else:
+                raise e
+
```

Comment:
Same as mentioned above

Reply:
Done in https://github.com/red-hat-storage/noobaa-sa-ci/pull/16/commits/cde85fc0c8aae17993097f4a72b721a00cc42873